File: tablet/Resources/menu/GlobalMenu.js
// GlobalMenu.js

/*
    Copyright (C) 2012-2015 Rudolf Cardinal (rudolf@pobox.com).
    Department of Psychiatry, University of Cambridge.
    Funded by the Wellcome Trust.

    This file is part of CamCOPS.

    Licensed under the Apache License, Version 2.0 (the "License");
    you may not use this file except in compliance with the License.
    You may obtain a copy of the License at

        http://www.apache.org/licenses/LICENSE-2.0

    Unless required by applicable law or agreed to in writing, software
    distributed under the License is distributed on an "AS IS" BASIS,
    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
    See the License for the specific language governing permissions and
    limitations under the License.
*/

/*jslint node: true, newcap: true */
"use strict";
/*global L */

module.exports = function GlobalMenu() {

    var MenuWindow = require('menulib/MenuWindow'),
        UICONSTANTS = require('common/UICONSTANTS'),
        ALLTASKS = require('common/ALLTASKS'),
        data = [
            UICONSTANTS.CHANGE_PATIENT_MENU_LINE,
            ALLTASKS.TASKLIST.BPRS,
            ALLTASKS.TASKLIST.BPRSE,
            ALLTASKS.TASKLIST.CBI_R,
            ALLTASKS.TASKLIST.CGI,
            ALLTASKS.TASKLIST.CGI_I,
            ALLTASKS.TASKLIST.COPE_BRIEF,
            ALLTASKS.TASKLIST.DEMQOL,
            ALLTASKS.TASKLIST.DEMQOL_PROXY,
            ALLTASKS.TASKLIST.DISTRESSTHERMOMETER,
            ALLTASKS.TASKLIST.GAF,
            ALLTASKS.TASKLIST.HONOS,
            ALLTASKS.TASKLIST.HONOS65,
            ALLTASKS.TASKLIST.HONOSCA,
            ALLTASKS.TASKLIST.RAND36,
            ALLTASKS.TASKLIST.SWEMWBS,
            ALLTASKS.TASKLIST.WEMWBS,
            ALLTASKS.TASKLIST.ZBI12,
        ],
        self = new MenuWindow({
            title: L('menutitle_global_b'),
            icon: UICONSTANTS.ICON_MENU_GLOBAL,
            data: data,
        });

    return self;
};

File: tablet_qt/menu/globalmenu.cpp
/*
    Copyright (C) 2012, University of Cambridge, Department of Psychiatry.
    Created by Rudolf Cardinal (rnc1001@cam.ac.uk).

    This file is part of CamCOPS.

    CamCOPS is free software: you can redistribute it and/or modify
    it under the terms of the GNU General Public License as published by
    the Free Software Foundation, either version 3 of the License, or
    (at your option) any later version.

    CamCOPS is distributed in the hope that it will be useful,
    but WITHOUT ANY WARRANTY; without even the implied warranty of
    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
    GNU General Public License for more details.

    You should have received a copy of the GNU General Public License
    along with CamCOPS. If not, see <https://www.gnu.org/licenses/>.
*/

#include "globalmenu.h"

#include "common/uiconst.h"
#include "lib/uifunc.h"
#include "menulib/menuitem.h"
#include "tasks/badls.h"
#include "tasks/bprs.h"
#include "tasks/bprse.h"
#include "tasks/cbir.h"
#include "tasks/cgi.h"
#include "tasks/cgii.h"
#include "tasks/copebrief.h"
#include "tasks/dad.h"
#include "tasks/demqol.h"
#include "tasks/demqolproxy.h"
#include "tasks/distressthermometer.h"
#include "tasks/eq5d5l.h"
#include "tasks/factg.h"
#include "tasks/frs.h"
#include "tasks/gaf.h"
#include "tasks/honos.h"
#include "tasks/honos65.h"
#include "tasks/honosca.h"
#include "tasks/npiq.h"
#include "tasks/ors.h"
#include "tasks/paradise24.h"
#include "tasks/rand36.h"
#include "tasks/swemwbs.h"
#include "tasks/wemwbs.h"
#include "tasks/wsas.h"
#include "tasks/zbi12.h"

GlobalMenu::GlobalMenu(CamcopsApp& app) :
    MenuWindow(app, uifunc::iconFilename(uiconst::ICON_GLOBAL))
{
}

QString GlobalMenu::title() const
{
    return tr("Global function and multiple aspects of psychopathology");
}

void GlobalMenu::makeItems()
{
    m_items = {
        MAKE_CHANGE_PATIENT(m_app),
        MAKE_TASK_MENU_ITEM(Badls::BADLS_TABLENAME, m_app),
        MAKE_TASK_MENU_ITEM(Bprs::BPRS_TABLENAME, m_app),
        MAKE_TASK_MENU_ITEM(BprsE::BPRSE_TABLENAME, m_app),
        MAKE_TASK_MENU_ITEM(CbiR::CBIR_TABLENAME, m_app),
        MAKE_TASK_MENU_ITEM(Cgi::CGI_TABLENAME, m_app),
        MAKE_TASK_MENU_ITEM(CgiI::CGI_I_TABLENAME, m_app),
        MAKE_TASK_MENU_ITEM(CopeBrief::COPEBRIEF_TABLENAME, m_app),
        MAKE_TASK_MENU_ITEM(Dad::DAD_TABLENAME, m_app),
        MAKE_TASK_MENU_ITEM(Demqol::DEMQOL_TABLENAME, m_app),
        MAKE_TASK_MENU_ITEM(DemqolProxy::DEMQOLPROXY_TABLENAME, m_app),
        MAKE_TASK_MENU_ITEM(DistressThermometer::DT_TABLENAME, m_app),
        MAKE_TASK_MENU_ITEM(Eq5d5l::EQ5D5L_TABLENAME, m_app),
        MAKE_TASK_MENU_ITEM(Factg::FACTG_TABLENAME, m_app),
        MAKE_TASK_MENU_ITEM(Frs::FRS_TABLENAME, m_app),
        MAKE_TASK_MENU_ITEM(Gaf::GAF_TABLENAME, m_app),
        MAKE_TASK_MENU_ITEM(Honos::HONOS_TABLENAME, m_app),
        MAKE_TASK_MENU_ITEM(Honos65::HONOS65_TABLENAME, m_app),
        MAKE_TASK_MENU_ITEM(Honosca::HONOSCA_TABLENAME, m_app),
        MAKE_TASK_MENU_ITEM(Ors::ORS_TABLENAME, m_app),
        MAKE_TASK_MENU_ITEM(NpiQ::NPIQ_TABLENAME, m_app),
        MAKE_TASK_MENU_ITEM(Paradise24::PARADISE24_TABLENAME, m_app),
        MAKE_TASK_MENU_ITEM(Rand36::RAND36_TABLENAME, m_app),
        MAKE_TASK_MENU_ITEM(Swemwbs::SWEMWBS_TABLENAME, m_app),
        MAKE_TASK_MENU_ITEM(Wemwbs::WEMWBS_TABLENAME, m_app),
        MAKE_TASK_MENU_ITEM(Wsas::WSAS_TABLENAME, m_app),
        MAKE_TASK_MENU_ITEM(Zbi12::ZBI12_TABLENAME, m_app),
    };
}
